base/inference.jl

```diff
@@ -4462,10 +4462,18 @@ function inlining_pass(e::Expr, sv::InferenceState, stmts, ins)
                 elseif isa(aarg, Tuple)
                     newargs[i-2] = Any[ QuoteNode(x) for x in aarg ]
                 elseif isa(t, DataType) && t.name === Tuple.name && !isvatuple(t) &&
-                        effect_free(aarg, sv.src, sv.mod, true) && length(t.parameters) <= sv.params.MAX_TUPLE_SPLAT
-                    # apply(f,t::(x,y)) => f(t[1],t[2])
+                         length(t.parameters) <= sv.params.MAX_TUPLE_SPLAT
+                    if effect_free(aarg, sv.src, sv.mod, true)
+                        # apply(f,t::(x,y)) => f(t[1],t[2])
+                        tmpv = aarg
+                    else
+                        # apply(f,t::(x,y)) => tmp=t; f(tmp[1],tmp[2])
+                        tmpv = add_slot!(sv.src, t, false)
```

Comment:
I think this should use `newvar!` and make an SSAValue instead. (That function is badly named; should be e.g. `new_ssavalue`.)